pkg/util/validation/limits_test.go

```diff
@@ -668,20 +668,35 @@ func TestEnabledByAnyTenant(t *testing.T) {
 	require.True(t, EnabledByAnyTenant([]string{"tenant1", "tenant2", "tenant3"}, ov.NativeHistogramsIngestionEnabled))
 }
 
+type structExtension struct {
+	Foo int `yaml:"foo"`
+}
+
+func (te *structExtension) SetDefaults() {
+	te.Foo = 42
+}
+
+type stringExtension string
+
+func (s *stringExtension) SetDefaults() {
+	*s = "default"
+}
+
+type nonPtrExtension string
+
+func (nonPtrExtension) SetDefaults() {
+	// do nothing
+}
+
 func TestExtensions(t *testing.T) {
 	t.Cleanup(func() {
 		registeredExtensionsIndexes = map[string]int{}
 		limitsExtensionsFields = nil
```

Comment:
should this also clear `extensionIsPointer`?

Alternatively, you can unify `extensionIsPointer` and `registeredExtensionsIndexes` into something like 

```go

type extensionIndex struct { 
	idx   int
	isPtr bool
}

var extentionIndexes = map[string]extensionIndex{}

```